ElasticSearch TestCreatesRoleBinding sometimes fails

```
...
go test $(go list ./... | grep -v '/vendor/')
?   	github.com/jetstack/navigator/cmd/apiserver	[no test files]
?   	github.com/jetstack/navigator/cmd/controller	[no test files]
?   	github.com/jetstack/navigator/cmd/controller/app	[no test files]
?   	github.com/jetstack/navigator/cmd/controller/app/options	[no test files]
?   	github.com/jetstack/navigator/cmd/pilot-elasticsearch	[no test files]
?   	github.com/jetstack/navigator/pkg/apis/navigator	[no test files]
?   	github.com/jetstack/navigator/pkg/apis/navigator/install	[no test files]
?   	github.com/jetstack/navigator/pkg/apis/navigator/v1alpha1	[no test files]
?   	github.com/jetstack/navigator/pkg/apiserver	[no test files]
?   	github.com/jetstack/navigator/pkg/client/clientset/internalversion	[no test files]
?   	github.com/jetstack/navigator/pkg/client/clientset/internalversion/fake	[no test files]
?   	github.com/jetstack/navigator/pkg/client/clientset/internalversion/scheme	[no test files]
?   	github.com/jetstack/navigator/pkg/client/clientset/internalversion/typed/navigator/internalversion	[no test files]
?   	github.com/jetstack/navigator/pkg/client/clientset/internalversion/typed/navigator/internalversion/fake	[no test files]
?   	github.com/jetstack/navigator/pkg/client/clientset/versioned	[no test files]
?   	github.com/jetstack/navigator/pkg/client/clientset/versioned/fake	[no test files]
?   	github.com/jetstack/navigator/pkg/client/clientset/versioned/scheme	[no test files]
?   	github.com/jetstack/navigator/pkg/client/clientset/versioned/typed/navigator/v1alpha1	[no test files]
?   	github.com/jetstack/navigator/pkg/client/clientset/versioned/typed/navigator/v1alpha1/fake	[no test files]
?   	github.com/jetstack/navigator/pkg/client/informers/externalversions	[no test files]
?   	github.com/jetstack/navigator/pkg/client/informers/externalversions/internalinterfaces	[no test files]
?   	github.com/jetstack/navigator/pkg/client/informers/externalversions/navigator	[no test files]
?   	github.com/jetstack/navigator/pkg/client/informers/externalversions/navigator/v1alpha1	[no test files]
?   	github.com/jetstack/navigator/pkg/client/informers/internalversion	[no test files]
?   	github.com/jetstack/navigator/pkg/client/informers/internalversion/internalinterfaces	[no test files]
?   	github.com/jetstack/navigator/pkg/client/informers/internalversion/navigator	[no test files]
?   	github.com/jetstack/navigator/pkg/client/informers/internalversion/navigator/internalversion	[no test files]
?   	github.com/jetstack/navigator/pkg/client/listers/navigator/internalversion	[no test files]
?   	github.com/jetstack/navigator/pkg/client/listers/navigator/v1alpha1	[no test files]
?   	github.com/jetstack/navigator/pkg/controllers	[no test files]
?   	github.com/jetstack/navigator/pkg/controllers/elasticsearch	[no test files]
?   	github.com/jetstack/navigator/pkg/controllers/elasticsearch/configmap	[no test files]
?   	github.com/jetstack/navigator/pkg/controllers/elasticsearch/nodepool	[no test files]
ok  	github.com/jetstack/navigator/pkg/controllers/elasticsearch/role	0.079s
--- FAIL: TestCreatesRoleBinding (0.01s)
	controller_test.go:103: Expected
			testing.CreateActionImpl{ActionImpl:testing.ActionImpl{Namespace:"default", Verb:"create", Resource:schema.GroupVersionResource{Group:"", Version:"", Resource:"rolebindings"}, Subresource:""}, Name:"", Object:(*v1beta1.RoleBinding)(0xc4200a6000)}
		got
			testing.ListActionImpl{ActionImpl:testing.ActionImpl{Namespace:"", Verb:"list", Resource:schema.GroupVersionResource{Group:"rbac.authorization.k8s.io", Version:"v1beta1", Resource:"rolebindings"}, Subresource:""}, Kind:schema.GroupVersionKind{Group:"rbac.authorization.k8s.io", Version:"v1beta1", Kind:"RoleBinding"}, Name:"", ListRestrictions:testing.ListRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}}}
	controller_test.go:97: 2 unexpected actions: [{ActionImpl:{Namespace: Verb:watch Resource:{Group:rbac.authorization.k8s.io Version:v1beta1 Resource:rolebindings} Subresource:} WatchRestrictions:{Labels: Fields: ResourceVersion:}} {ActionImpl:{Namespace:default Verb:create Resource:{Group:rbac.authorization.k8s.io Version:v1beta1 Resource:rolebindings} Subresource:} Name: Object:&RoleBinding{ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{Name:es-test,GenerateName:,Namespace:default,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{app: elasticsearch,navigator.jetstack.io/elasticsearch-cluster-name: test,},Annotations:map[string]string{},OwnerReferences:[{navigator.jetstack.io/v1alpha1 ElasticsearchCluster test 719795ca-c478-11e7-ae11-42010a14003d 0xc4203460f8 0xc4203460f7}],Finalizers:[],ClusterName:,Initializers:nil,},Subjects:[{ServiceAccount  es-test }],RoleRef:RoleRef{APIGroup:,Kind:Role,Name:es-test,},}}]
FAIL
FAIL	github.com/jetstack/navigator/pkg/controllers/elasticsearch/rolebinding	0.122s
?   	github.com/jetstack/navigator/pkg/controllers/elasticsearch/service	[no test files]
?   	github.com/jetstack/navigator/pkg/controllers/elasticsearch/serviceaccount	[no test files]
?   	github.com/jetstack/navigator/pkg/controllers/elasticsearch/util	[no test files]
?   	github.com/jetstack/navigator/pkg/pilot/elasticsearch/v5	[no test files]
?   	github.com/jetstack/navigator/pkg/pilot/genericpilot	[no test files]
?   	github.com/jetstack/navigator/pkg/pilot/genericpilot/hook	[no test files]
?   	github.com/jetstack/navigator/pkg/pilot/genericpilot/probe	[no test files]
?   	github.com/jetstack/navigator/pkg/pilot/genericpilot/process	[no test files]
?   	github.com/jetstack/navigator/pkg/registry	[no test files]
?   	github.com/jetstack/navigator/pkg/registry/navigator/escluster	[no test files]
?   	github.com/jetstack/navigator/pkg/registry/navigator/pilot	[no test files]
?   	github.com/jetstack/navigator/pkg/util	[no test files]
?   	github.com/jetstack/navigator/pkg/util/errors	[no test files]
ok  	github.com/jetstack/navigator/pkg/util/hash	0.026s
?   	github.com/jetstack/navigator/third_party/k8s.io/client-go/informers/externalversions	[no test files]
?   	github.com/jetstack/navigator/third_party/k8s.io/client-go/informers/externalversions/apps	[no test files]
?   	github.com/jetstack/navigator/third_party/k8s.io/client-go/informers/externalversions/apps/v1beta1	[no test files]
?   	github.com/jetstack/navigator/third_party/k8s.io/client-go/informers/externalversions/core	[no test files]
?   	github.com/jetstack/navigator/third_party/k8s.io/client-go/informers/externalversions/core/v1	[no test files]
?   	github.com/jetstack/navigator/third_party/k8s.io/client-go/informers/externalversions/internalinterfaces	[no test files]
?   	github.com/jetstack/navigator/third_party/k8s.io/client-go/informers/externalversions/rbac	[no test files]
?   	github.com/jetstack/navigator/third_party/k8s.io/client-go/informers/externalversions/rbac/v1beta1	[no test files]
make: *** [go_test] Error 1
```
https://travis-ci.org/jetstack/navigator/jobs/299051804#L534

[log.txt](https://github.com/jetstack/navigator/files/1453703/log.txt)

/kind bug

--- a/pkg/controllers/elasticsearch/role/controller_test.go
+++ b/pkg/controllers/elasticsearch/role/controller_test.go
@@ -114,10 +114,8 @@ func filterInformerActions(actions []core.Action) []core.Action {
 	ret := []core.Action{}
 	for _, action := range actions {
 		if len(action.GetNamespace()) == 0 &&
-			(action.Matches("list", "foos") ||
-				action.Matches("watch", "foos") ||
-				action.Matches("list", "deployments") ||
-				action.Matches("watch", "deployments")) {
+			(action.Matches("list", "roles") ||
+				action.Matches("watch", "roles")) {
 			continue
 		}
 		ret = append(ret, action)

